samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -315,4 +315,48 @@ public void onBarrierError(String version, Throwable t) {
       stop();
     }
   }
+
+  /// listener to handle session expiration
+  public class ZkSessionStateChangedListener implements IZkStateListener {
```

Comment:
yes. Functionally this class makes sense only for the JobCoordinator, so I didn't see a reason to take it out.